apps/github-stack/src/github-ineraction/domain/scheduler/github-scheduler.ts

```diff
@@ -1,18 +1,80 @@
-import { Injectable } from '@nestjs/common'
-import { Cron, CronExpression } from '@nestjs/schedule'
-import { SendingEmailService } from '../../service/sending-email.service'
+import { Injectable } from "@nestjs/common";
+import { Cron, CronExpression } from "@nestjs/schedule";
+import { SendingEmailService } from "../../service/sending-email.service";
+import { HttpService } from "@nestjs/axios";
+import { ConfigService } from "@nestjs/config";
+import { InjectRepository } from "@nestjs/typeorm";
+import { firstValueFrom } from "rxjs";
+import { Repository } from "typeorm";
+import { User } from "../../../users/domain/entity/user.entity";
+import { GitRepository } from "../entity/repository.entity";
 
 @Injectable()
 export class GitHubScheduler {
-  constructor(private readonly sendingEmailService: SendingEmailService) {}
+  constructor(
+    private readonly httpService: HttpService,
+    private readonly configService: ConfigService,
+    private readonly sendingEmailService: SendingEmailService,
+    @InjectRepository(User)
+    private readonly userRep: Repository<User>,
+    @InjectRepository(GitRepository)
+    private readonly gitRepository: Repository<GitRepository>
+  ) {}
+
+  private readonly githubApiUrl = "https://api.github.com"
 
   @Cron(CronExpression.EVERY_10_SECONDS)
   async handleCron() {
-    await this.sendingEmailService.checkForUpdates()
+    await this.checkForUpdates();
```

Comment:
checkForUpdates should return boolean value and if it is true you can proceed to send the given email.